crates/datastore/src/locking_tx_datastore/mut_tx.rs

```diff
@@ -686,6 +730,26 @@ impl MutTxId {
 
         Ok((tx, commit))
     }
+
+    /// Check if a memoized view exists for the given view name, args, and sender identity.
+    /// if not, [`RelationalDB::evaluate_view`] should be called to compute and store it.
+    fn has_memoized_view(&self, view_name: &str, args: Bytes, sender: Identity) -> Result<(bool, Bytes)> {
```

Comment:
I think this should be implemented as `impl Datastore`, Query engine if contains a View in plan should call this method to check if `call_view` should be invoked.